WearSpeakerSample/wear/src/main/java/com/example/android/wearable/speaker/MainActivity.kt

```diff
@@ -0,0 +1,269 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.speaker
+
+import android.Manifest
+import android.app.AlertDialog
+import android.content.pm.PackageManager
+import android.media.MediaPlayer
+import android.os.Bundle
+import android.util.Log
+import androidx.activity.result.contract.ActivityResultContracts.RequestPermission
+import androidx.activity.viewModels
+import androidx.annotation.IdRes
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.example.android.wearable.speaker.MainViewModel.AppAction
+import com.example.android.wearable.speaker.MainViewModel.AppState
+import com.example.android.wearable.speaker.databinding.MainActivityBinding
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.suspendCancellableCoroutine
+import java.time.Duration
+import kotlin.coroutines.resume
+
+/**
+ * We first get the required permission to use the MIC. If it is granted, then we continue with
+ * the application and present the UI with three icons: a MIC icon (if pressed, user can record up
+ * to 10 seconds), a Play icon (if clicked, it will playback the recorded audio file) and a music
+ * note icon (if clicked, it plays an MP3 file that is included in the app).
+ */
+class MainActivity : AppCompatActivity() {
+
+    private val viewModel by viewModels<MainViewModel>()
+
+    private lateinit var binding: MainActivityBinding
+
+    private var soundRecorder: SoundRecorder? = null
+
+    private var ongoingWork: Job = Job().apply { complete() }
+
+    private val requestPermissionLauncher = registerForActivityResult(RequestPermission()) {
+        // We ignore the direct result here, since we're going to check anyway.
+        viewModel.onAction(AppAction.PermissionResultReturned)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = MainActivityBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.mic.setOnClickListener {
+            viewModel.onAction(
+                AppAction.MicClicked(shouldShowRequestPermissionRationale(Manifest.permission.RECORD_AUDIO))
+            )
+        }
+        binding.play.setOnClickListener { viewModel.onAction(AppAction.PlayClicked) }
+        binding.music.setOnClickListener { viewModel.onAction(AppAction.MusicClicked) }
+
+        lifecycleScope.launch {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.appState
+                    .onEach { appState ->
```

Comment:
Nope, there should be no difference with this usage.

It might just be a personal preference, but using `onEach` keeps consistent usage regardless of whether `launchIn(...)` or `collect()` is used, and if a `catch { ... }` block is necessary to catch an exception thrown from `onEach`.